packages/flutter_sprites/lib/src/sound.dart

```diff
@@ -1,139 +1,111 @@
-    this.sound,
-    this.loop,
-    this.volume,
-    this.pitch,
-    this.pan,
-    this.onSoundComplete
-  );
-
-  // TODO: Make these properties work
-  SoundEffect sound;
-  bool playing = false;
-  bool loop = false;
-  double volume = 1.0;
-  double pitch = 1.0;
-  double pan = 0.0;
-
-  // TODO: Implement completion callback. On completion, sounds should
-  // also be removed from the list of playing sounds.
-  SoundEffectStreamCallback onSoundComplete;
+  int _streamId;
+  bool _paused;
+  bool _loop;
+  double _leftVolume;
+  double _rightVolume;
+  double _pitch;
 
-  MediaPlayerProxy _player;
-}
+  SoundEffectStream(this._streamId, this._leftVolume, this._rightVolume,
+                    this._loop, this._pitch);
 
-SoundEffectPlayer _sharedSoundEffectPlayer;
+  SoundPoolProxy get _soundPool =>
+      SoundEffectPlayer.sharedInstance()._soundPool;
 
-class SoundEffectPlayer {
+  void stop() {
+    _soundPool.ptr.stop(_streamId);
+  }
 
-  static SoundEffectPlayer sharedInstance() {
-    if (_sharedSoundEffectPlayer == null) {
-      _sharedSoundEffectPlayer = new SoundEffectPlayer();
+  bool get paused => _paused;
+  void set paused(bool value) {
+    _paused = value;
+    if (_paused) {
+      _soundPool.ptr.pause(_streamId);
+    } else {
+      _soundPool.ptr.resume(_streamId);
     }
-    return _sharedSoundEffectPlayer;
   }
 
-  SoundEffectPlayer() {
-    _mediaService = new MediaServiceProxy.unbound();
-    shell.connectToService(null, _mediaService);
+  double get leftVolume => _leftVolume;
+  void set leftVolume(double value) {
+    _leftVolume = value;
+    _soundPool.ptr.setVolume(_streamId, _leftVolume, _rightVolume);
   }
 
-  MediaServiceProxy _mediaService;
-  List<SoundEffectStream> _soundEffectStreams = <SoundEffectStream>[];
-
-  // TODO: This should no longer be needed when moving to SoundPool backing
-  Map<SoundEffect,MediaPlayerProxy> _mediaPlayers = <SoundEffect, MediaPlayerProxy>{};
+  double get rightVolume => _rightVolume;
+  void set rightVolume(double value) {
+    _rightVolume = value;
+    _soundPool.ptr.setVolume(_streamId, _leftVolume, _rightVolume);
+  }
 
-  Future _prepare(SoundEffectStream playingSound) async {
-    await playingSound._player.ptr.prepare(playingSound.sound._data);
+  double get pitch => _pitch;
+  void set pitch(double value) {
+    _pitch = value;
+    _soundPool.ptr.setRate(_streamId, _pitch);
   }
+}
 
-  // TODO: Move sound loading here
-  // TODO: Support loading sounds from bundles
-  // Future<SoundEffect> load(url) async {
-  //   ...
-  // }
+class SoundEffectPlayer {
+  SoundPoolProxy _soundPool;
+  bool _paused;
 
-  // TODO: Add sound unloader
-  // unload(SoundEffect effect) {
-  //   ...
-  // }
+  static const int _kMaxStreams = 20;
 
-  // TODO: Add paused property (should pause playback of all sounds)
-  bool paused;
+  static final SoundEffectPlayer _instance = new SoundEffectPlayer();
+  static SoundEffectPlayer sharedInstance() => _instance;
```

Comment:
Why is sharedInstance() a method, instead of just having a public getter "instance"?